README.md

```diff
@@ -1,368 +1,257 @@
 
-### Generated JavaScript
+`react_component(component_name, props = {}, options = {})`
++ **react_component_name:** Can be a React component, created using a ES6 class, or `React.createClass`, or a generator function that returns a React component.
++ **props:** Ruby Hash which contains the properties to pass to the react object
++ **options:**
+  + **generator_function:** default is false, set to true if you want to use a generator function rather than a React Component.
+  + **prerender:** enable server-side rendering of component. Set to false when debugging!
+  + **trace:** set to true to print additional debugging information in the browser. Defaults to true for development, off otherwise.
+  + **replay_console:** Default is true. False will disable echoing server-rendering logs to the browser. While this can make troubleshooting server rendering difficult, so long as you have the default configuration of logging_on_server set to true, you'll still see the errors on the server.
++ Any other options are passed to the content tag, including the id
 
-1. See [sample_generated_js/server-generated.js](docs/sample_generated_js/server-generated.js) to see the JavaScript for typical server rendering.
-2. See [sample_generated_js/client-generated.js](docs/sample_generated_js/client-generated.js) to see the JavaScript for typical client rendering.
+`def server_render_js(js_expression, options = {})`
 
-## Contributing
+This is a helper method that takes any JavaScript expression and returns the output from evaluating it. If you have more than one line that needs to be executed, wrap it in an IIFE. JS exceptions will be caught and console messages handled properly.
 
-Bug reports and pull requests are welcome on GitHub at https://github.com/shakacode/react_on_rails. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](contributor-covenant.org) code of conduct.
+## Developing with Webpack Dev Server
+While developing your client code, you *could* simply re-build anytime you make a change, boot up your Rails server, and view the result. However, you would be missing out on the benefits of [webpack's dev server](https://webpack.github.io/docs/webpack-dev-server.html) and its [hot module replacement](https://webpack.github.io/docs/hot-module-replacement-with-webpack.html) functionality.
 
-More [tips on contributing here](docs/Contributing.md)
+Instead of going through Rails, you can use the generated `client/server.js` to start up the dev server and load the provided `client/index.jade`. (You will need to configure this file to load your client app inside of a `script` tag.) Then, all you need to do is run the build script and start the server:
 
-## License
++ *Normal Mode (JavaScript is Rendered on client):*
```

Comment:
Yes totally agreed, users should just use the Procfile.dev, this is TMI and not necessary, removed